stdlib/public/core/String.swift

```diff
@@ -867,36 +872,41 @@ extension String {
 
     // TODO(String performance): Try out incremental case-conversion rather than
     // make UTF-16 array beforehand
+    enum DummyError: Error { case dummy }
     let codeUnits = Array(self.utf16).withUnsafeBufferPointer {
       (uChars: UnsafeBufferPointer<UInt16>) -> Array<UInt16> in
-      var result = Array<UInt16>(repeating: 0, count: uChars.count)
-      let len = result.withUnsafeMutableBufferPointer {
-        (output) -> Int in
+      var len: Int = 0
+      let result = try? Array<UInt16>(unsafeUninitializedCapacity: uChars.count) {
+        buf, initializedCount in
         var err = __swift_stdlib_U_ZERO_ERROR
-        return Int(truncatingIfNeeded:
+        len = Int(truncatingIfNeeded:
           __swift_stdlib_u_strToUpper(
-            output.baseAddress._unsafelyUnwrappedUnchecked,
-            Int32(output.count),
+            buf.baseAddress._unsafelyUnwrappedUnchecked,
+            Int32(buf.count),
```

Comment:
Nit: In general, the stdlib code base avoids ad-hoc abbreviations such as “buf” or “ret”; I realize some of these were not created by you, but ideally we wouldn’t introduce more.

Reply:
Renamed variables.